cranelift-codegen/meta/src/shared/instructions.rs

```diff
@@ -1685,6 +1685,38 @@ pub(crate) fn define(
         .operands_out(vec![a]),
     );
 
+    ig.push(
+        Inst::new(
+            "uadd_sat",
```

Comment:
Can these new instructions (`*_sat`, `ineg`) have non simd encodings too? That would make cg_clif easier and codegen better.

Reply:
`ineg` could use x86 `NEG` but I think saturating add/sub would have to be legalized (e.g. https://stackoverflow.com/a/166393). This could be done in a separate PR, though.

Reply:
You think I should add the `NEG` encoding to this one?

Reply:
Yes please.

Reply:
He, easier said than done apparently. The `NEG` instructions wants to insert `3` in the reg field of the ModR/M byte. Not hard, right? But the recipe/template distinction and `PerCpuModeEncodings` make it hard to put this `3` in `EncodingBuilder::new(..., ..., bits)`. Do you see a good way to do this? I've been thinking about refactor `PerCpuModeEncodings`...

Reply:
I am not familiar with `PerCpuEncodings`. It's fine to leave it out of this PR if it is hard to add. I am currently using `v1 = iconst 0` + `isub v1, v0` in cg_clif.

Reply:
Ok, let me add it as an issue: https://github.com/CraneStation/cranelift/issues/1053.